### DISCUSSION TITLE
Nuxt 3

### DISCUSSION THREAD

**Discussion by Triloworld** [ivodolenc/nuxt-font-loader#8] [General]
Is this planned?

What can be added in new version as well as thinks that in scope of module can improve experience:
- typescript
- tree shaking for less js on frontend
- hooks for other modules (we plan wordpress integration, right now you have only 'build:before' use but no new one introduce to inject something externall
- auto import some fonts (Google etc) from tailwind/wind config file (this can be cool feature btw)
- better default options
- webpack 5/Vite support
- directive ? 
- incorporate better console logs and give this option to silent on ci or test (https://github.com/unjs/consola)

**Comment by ivodolenc** [comment]
Hi, sorry for the long wait but I wanted to wait for the stable Nuxt 3 version and see what the new API brings.

There are a lot of new and improved features in Nuxt 3, so I've updated the module accordingly. 

Also, I'm working on a new `external` feature, so it will be possible to upload fonts via an alternative, i.e. via a third-party server like Google or Typekit (need some time to finish the types and documentation).

I added typescript support and simplified the API.

Here are simple example:

```js
// nuxt.config.ts

{
  modules: ['nuxt-font-loader'],

  fontLoader: {
    local: [
      {
        src: '/fonts/AspektaVF.woff2',
        family: 'Aspekta Variable',
        weight: '100 900'
      }
    ]
  }
}
```

Feel free to try it out and leave feedback.
